src/plugins/intel_gpu/src/kernel_selector/kernels/rms/rms_kernel_bfyx_opt.cpp

```diff
@@ -58,9 +60,9 @@ JitConstants RMSKernelBfyxOpt::GetJitConstants(const rms_params& params, Dispatc
         }
 
         const std::string lws_0 = "get_local_size(0)";
-        // It can be expected that the maximum possible itemsNum will not exceed 32
-        // Therefore, in dynamic shape, stack_size including additional buffer is set to 33
-        constexpr size_t stack_size = 33;
+        // It can be expected that the maximum possible itemsNum will not exceed MAX_ITEMS_NUM
+        // Therefore, in dynamic shape, stack_size including additional buffer is set to MAX_ITEMS_NUM
+        constexpr size_t stack_size = MAX_ITEMS_NUM;
```

Comment:
Have you accounted for the fact that we may have leftovers?

A thing you could do is that if data_set_size is a multiple of the maximum LWS then you don't have to include the leftover in the stack size aswell, since it would be impossible for any LWS to have leftovers. This is only for dynamic case

Also, there are some changes in the way items_num and leftovers are calculated in the RMS kernel that are on the queue. Please rebase your development environment to master after it's merged

Reply:
Therefore my suggestion is that
If dataSize is not a multiple of max LWS size, theoretically we could have a situation where we have leftovers, so we set stack_size to MAX_ITEMS_NUM + 1

If dataSize is a multiple of max LWS size (and there are plenty such cases actually) then set stack_size to MAX_ITEMS_NUM, as it is not possible for any LWS (that is constrained to be a power of two) to leave any leftovers when calculating items_num

Reply:
Mentioned PR is as follows https://github.com/openvinotoolkit/openvino/pull/26495

Also, nice find! Thank you!

Reply:
Also, theoretically it could be possible for dynamic shapes to actually have a larger itemsNum than 8, and I guess it would be incorrect then. Probably you could use an if inside RMS kernel to disable usage of private memory if private memory storage would exceed this MAX_ITEMS_NUM

Reply:
I think items_num is always smaller than 8 because of this line https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/kernel_selector/kernels/rms/rms_kernel_bfyx_opt.cpp#L123 . If the stack size is set to 8, it will cover the leftovers.

Reply:
Not really since if you have a large enough itemsNum then the && (2 * dispatchData.lws[0] <= max_lws) condition would be false, since lws increases by double per each step. It could leave you with a itemsNum bigger than 8

Reply:
So I guess the idea would be to maybe set stack size to MAX_ITEMS_NUM + 1, and have a small if inside kernel to not use private memory for caching

Reply:
softmax bf opt kernel does a similar thing, you may take a look at it and see how it's done in softmax

Reply:
And also it could be good to check if bigger LLM models and other dynamic networks that use RMS could actually slow down.

Reply:
> Actually, you could calculate a minimum and a maximum stack size you would need, based on data_size which is indeed constant, since you could have a minimum data count size which could give you maximum items_num

Thanks for your suggestion. I added a simple condition so that I didn’t have to change too much code.

Reply:
That is only valid if max_lws is >= 512

Reply:
But if dataset is constant then you might aswell give it itemsNum as a stack size. For qwen 0.5B I calculated it would always require a stack size of 7, without leftovers, which actually would be optimal. And with a LWS of 128

Reply:
The following code works right for me for qwen2 0.5B, and it correctly sets stack size to 7 which is better than your estimate of 8 + 1

`    if (params.has_dynamic_tensors()) {
        const auto& input = params.inputs[0];
        DimensionAccessHelperJit dims(input);
        std::string data_size;
        switch (params.ov_input_rank) {
            case 1 :
                data_size = dims.b();
                break;
            case 2 :
                data_size = dims.f();
                break;
            case 3 :
                data_size = dims.y();
                break;
            default:
                data_size = dims.x();
                break;
        }

        unsigned int lws = 1;
        unsigned int itemsNum = std::stoi(data_size);
        auto local_mem_per_wi = 2 * BytesPerElement(input.GetDType());
        auto max_lws = std::min(params.engineInfo.maxWorkGroupSize, params.engineInfo.maxLocalMemSize / local_mem_per_wi);

        while ((itemsNum > 8 || lws < itemsNum) && (2 * lws <= max_lws)) {
            lws *= 2;
            itemsNum /= 2;
        }

        jit.AddConstants({
            MakeJitConstant("DATA_SIZE", data_size),
            MakeJitConstant("LWS", lws),
            MakeJitConstant("SLM_SIZE", dispatchData.maxSlmSize),
            MakeJitConstant("STACK_SIZE", cldnn::ceil_div(std::stoi(data_size), lws))
        });
`

Reply:
Thanks for your suggestion. The code has now been updated.

Reply:
Also, just curious, does the new approach have better performance than your previous approach? I think it should be a bit faster for qwen2 0.5B because now we are using stack size 7 instead of 8 which should be more efficient

Reply:
> Also, just curious, does the new approach have better performance than your previous approach? I think it should be a bit faster for qwen2 0.5B because now we are using stack size 7 instead of 8 which should be more efficient.
> 
> You may also update PR description too

No noticeable performance difference using stack size 7, 8, 9.